docs/references/sql-introspection.md

```diff
@@ -0,0 +1,129 @@
+
+## Table: `state`
+
+| Column name | Type | Description |
+|-------------|------|-------------|
+| `partition_key` | `UInt64` | Internal column that is used for partitioning the services invocations. Can be ignored. |
+| `service_name` | `Utf8` | The name of the invoked service. |
+| `service_key` | `Utf8` | The key of the Virtual Object. |
+| `key` | `Utf8` | The `utf8` state key. |
+| `value_utf8` | `Utf8` | Only contains meaningful values when a service stores state as `utf8`. This is the case for TypeScript services since the TypeScript SDK serializes values as JSON. |
+| `value` | `Binary` | A binary, uninterpreted representation of the value. You can use the more specific column `value_utf8` if the value is a string. |
+
+## Table: `sys_invocation`
+
+| Column name | Type | Description |
+|-------------|------|-------------|
+| `id` | `Utf8` | [Invocation ID](/operate/invocation#invocation-identifier). |
+| `target` | `Utf8` | Invocation Target. Format for plain services: `ServiceName/HandlerName`, e.g. `Greeter/greet`. Format for Virtual Objects/Workflows: `VirtualObjectName/Key/HandlerName`, e.g. `Greeter/Francesco/greet`. |
+| `target_service_name` | `Utf8` | The name for the invoked service. |
+| `target_service_key` | `Utf8` | The key of the Virtual Object or of the Workflow. Null for regular services. |
+| `target_handler_name` | `Utf8` | The invoked handler. |
+| `target_service_ty` | `Utf8` | The service type. Either `service` or `virtual_object` or `workflow`. |
+| `invoked_by` | `Utf8` | Either `ingress` if the service was invoked externally or `service` if the service was invoked by another Restate service. |
+| `invoked_by_service_name` | `Utf8` | The name of the invoking service. Or `null` if invoked externally. |
+| `invoked_by_id` | `Utf8` | The caller [Invocation ID](/operate/invocation#invocation-identifier) if the service was invoked by another Restate service. Or `null` if invoked externally. |
+| `invoked_by_target` | `Utf8` | The caller invocation target if the service was invoked by another Restate service. Or `null` if invoked externally. |
+| `pinned_deployment_id` | `Utf8` | The opaque service deployment ID that has been committed for this invocation; this is set after the first journal entry is stored for this invocation. |
+| `trace_id` | `Utf8` | The ID of the trace that is assigned to this invocation. Only relevant when tracing is enabled. |
+| `journal_size` | `UInt32` | The number of journal entries durably logged for this invocation. |
+| `created_at` | `Date64` | Timestamp indicating the start of this invocation. |
+| `modified_at` | `Date64` | Timestamp indicating the last state transition. For example, last time the status changed from `invoked` to `suspended`. |
+| `retry_count` | `UInt64` | The number of attempts since the last successful attempt of this invocation. Increments on start, so 2 or more means a failure occurred. |
```

Comment:
`since the last successful attempt`? Not sure what we mean by this. A successful attempt to me sounds as if the invocation finished successfully.

Reply:
Yeah, the wording is a bit confusing. What we mean by this is that we don't persist the invocation state across suspensions or system restarts. So it is not the global retry_count.

Reply:
I replaced it with:

```
/// The number of invocation attempts since the current leader started executing it. Increments
    /// on start, so a value greater than 1 means a failure occurred. Note: the value is not a
    /// global attempt counter across invocation suspensions and leadership changes.
```

Only slightly better :-(